### ISSUE TITLE
Add support for Open Beauty Facts, Open Pet Food Facts, Open Products Facts which now provide the latest APIs (incl. Knowledge Panels)

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-explorer#113]

**Comment by KarruHarin** [comment]
I want to first try to integrate OpenBeautyFacts Api and then move on to others .I would like to be assigned .

**Comment by teolemon** [comment]
![Image](https://github.com/user-attachments/assets/bd4133f8-d504-43de-b544-164b35b988b2)
Trying to enter a Cosmetic barcode does not work: https://world.openbeautyfacts.org/product/3245678674840/bain-de-bouche-fraicheur-intense-auchan
https://openfoodfacts-explorer.vercel.app/products/3245678674840

**Comment by KarruHarin** [comment]
![Image](https://github.com/user-attachments/assets/2b26ff41-26cf-48f1-ad38-01e10d442c2a)

@VaiTon some beauty products are listed & some are not ,does that mean there are some beauty products in the OFF data?
and can I get a little guidence on how to get started ? I am looking forward to add support for OBF :)

**Comment by VaiTon** [comment]
> does that mean there are some beauty products in the OFF data

Unfortunately, yes. AFAIK we're only querying the OFF endpoints, so no OBF data should be displayed.

> Can I get a little guidence on how to get started ? I am looking forward to add support for OBF

@teolemon do we have a way of querying if a certain barcode is of a certain type? (other than trying to request it and having an error in the response?)

**Comment by KarruHarin** [comment]
@teolemon I want to add a search option for OBF , like a small drop down menu to either select OFF or OBF , then they can search in that respectively . Is that okay ? or do you have any suggestions on how to implement it in search .

**Comment by Bhavishya2601** [comment]
@VaiTon @teolemon what we can do is first find it in open food facts and if it is not found then we can search it in open beauty facts (right now I tried with beauty API only), I think this method is better then giving a dropdown to the user to select OFF or OBF.
I tried implementing it, you can see below

https://github.com/user-attachments/assets/0df49b84-acc0-4042-a87b-d2e9c71e1104

It is not able to show the data because it is configured according to OFF

**Comment by BunnyKarru** [comment]
@Bhavishya2601 but consider this when you search for beauty products you want to get results from OBF , but there are some beauty products in OFF too.
For example when I search shampoo,I want results from OBF ,but if we follow the approach you said we would get results from OFF as there is some unnecessary beauty data in OFF .That's why I preferred a drop down for search .

**Comment by Bhavishya2601** [comment]
@BunnyKarru One of the approach can be we can first search in OBF and then in OFF, but a better solution would be to remove those unnecessary beauty data from OFF
If we talk about barcode, the user doesn't know by just the barcode if it is in OFF or OBF (like I showed it video above), and if the user chooses to open the camera and scan the barcode then we can't have him select from the dropdown at that time

**Comment by Bhavishya2601** [comment]
@BunnyKarru bro if the user searches for egg or something then he will get multiple products (either from OBF or OFF), he can browse through it, obv he doesn't want to browse through all the products (10-15 pages) to get his desired product, he will obv go for searching his product directly, and then this method will work
Bro are you working on this issue?

**Comment by teolemon** [comment]
Here's an example of what we did on the mobile app

![Image](https://github.com/user-attachments/assets/793f9048-d022-4894-a39b-d7a5b53f6aea)

**Comment by BunnyKarru** [comment]
@teolemon @Bhavishya2601 yeah this was my idea , I will complete the task mostly within two days and raise a PR.

**Comment by BunnyKarru** [comment]
> [@BunnyKarru](https://github.com/BunnyKarru) bro if the user searches for egg or something then he will get multiple products (either from OBF or OFF), he can browse through it, obv he doesn't want to browse through all the products (10-15 pages) to get his desired product, he will obv go for searching his product directly, and then this method will work Bro are you working on this issue?

@Bhavishya2601 ,yeah I am already working on this issue
